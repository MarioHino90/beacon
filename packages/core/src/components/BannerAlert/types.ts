import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
