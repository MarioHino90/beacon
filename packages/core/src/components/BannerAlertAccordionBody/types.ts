export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
