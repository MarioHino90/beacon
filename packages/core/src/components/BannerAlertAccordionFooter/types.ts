export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
