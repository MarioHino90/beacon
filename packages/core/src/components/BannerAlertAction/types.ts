import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
