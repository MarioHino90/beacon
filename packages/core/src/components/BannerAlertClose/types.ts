import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
