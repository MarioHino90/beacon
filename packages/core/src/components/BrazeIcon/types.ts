import React from "react";

export type GenericSpotIcon<Props> = React.LazyExoticComponent<{
  (props: Props): React.ReactElement;
  displayName: string;
}>;

export interface SpotIcons {
  "add-document": GenericSpotIcon<AddDocumentProps>;
  "add-image-or-document": GenericSpotIcon<AddImageOrDocumentProps>;
  "arrow-toggle": GenericSpotIcon<ArrowToggleProps>;
  "campaign-comparison": GenericSpotIcon<CampaignComparisonProps>;
  content: GenericSpotIcon<ContentProps>;
  "email-settings": GenericSpotIcon<EmailSettingsProps>;
  "empty-table": GenericSpotIcon<AddDocumentProps>;
  "funnel-report": GenericSpotIcon<FunnelReportProps>;
  mms: GenericSpotIcon<MMSProps>;
  preview: GenericSpotIcon<PreviewProps>;
  "retention-report": GenericSpotIcon<RetentionReportProps>;
  sms: GenericSpotIcon<SMSProps>;
}

export interface BrazeIconProps extends StyledBrazeIconProps {
  /**
   * icon to be displayed to user */
  name?: keyof SpotIcons;
}

export interface StyledBrazeIconProps
  extends React.HTMLAttributes<HTMLElement> {
  /**
   * child elements to be rendered inside the component tree */
  children?: React.ReactNode | React.ReactNode[];
}

export interface AddDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface AddImageOrDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ArrowToggleProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface CampaignComparisonProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ContentProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface FunnelReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface EmailSettingsProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface PreviewProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface MMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface RetentionReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface SMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}
