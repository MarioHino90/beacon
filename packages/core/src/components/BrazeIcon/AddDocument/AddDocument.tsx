import React from "react";
import classNames from "classnames";

import { AddDocumentProps } from "../types";

export const AddDocument = ({
  className,
  ...props
}: AddDocumentProps): React.ReactElement => (
  <svg
    {...props}
    className={classNames(["bcl-add-document bcl-empty-table-icon", className])}
    width="73"
    height="55"
    viewBox="0 0 73 55"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <mask id="path-1-inside-1" fill="white">
      <rect x="0.5" y="0.125" width="64.4661" height="46.6852" rx="1" />
    </mask>
    <rect
      x="0.5"
      y="0.125"
      width="64.4661"
      height="46.6852"
      rx="1"
      fill="white"
      stroke="#101B24"
      strokeWidth="4"
      mask="url(#path-1-inside-1)"
    />
    <path d="M6.87939 23.0833H58.5866" stroke="#101B24" strokeWidth="2" />
    <path d="M6.87939 28.9031H58.5866" stroke="#101B24" strokeWidth="2" />
    <path d="M6.87939 34.499H58.5866" stroke="#101B24" strokeWidth="2" />
    <path d="M6.87939 39.7593H58.5866" stroke="#101B24" strokeWidth="2" />
    <rect
      x="6.82349"
      y="6.35156"
      width="19.2503"
      height="5.37217"
      fill="#75DBE7"
    />
    <rect
      x="6.87939"
      y="14.2417"
      width="38.3886"
      height="3.58145"
      fill="#C4F0F5"
    />
    <circle
      cx="63.2873"
      cy="45.2082"
      r="8.2128"
      fill="#75DBE7"
      stroke="#101B24"
      strokeWidth="2"
    />
    <path d="M63.4221 40.3308V49.8146" stroke="#101B24" strokeWidth="2" />
    <path
      d="M68.1643 45.0725L58.6805 45.0725"
      stroke="#101B24"
      strokeWidth="2"
    />
  </svg>
);

AddDocument.displayName = "AddDocument";
