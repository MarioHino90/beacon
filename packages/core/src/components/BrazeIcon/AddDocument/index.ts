export { AddDocument } from "./AddDocument";

export type { AddDocumentProps } from "../types";
