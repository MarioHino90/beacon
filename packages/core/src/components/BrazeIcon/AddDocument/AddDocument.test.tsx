import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { AddDocumentProps } from "../types";

import { AddDocument } from "./AddDocument";

faker.seed(1604940599873);

describe("AddDocument", () => {
  let props: AddDocumentProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<AddDocument {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
