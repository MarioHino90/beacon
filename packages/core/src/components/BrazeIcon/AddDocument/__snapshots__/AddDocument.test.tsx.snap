// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDocument should match the snapshot 1`] = `
<svg
  className="bcl-add-document bcl-empty-table-icon Savings"
  fill="none"
  height="55"
  id="input"
  viewBox="0 0 73 55"
  width="73"
  xmlns="http://www.w3.org/2000/svg"
>
  <mask
    fill="white"
    id="path-1-inside-1"
  >
    <rect
      height="46.6852"
      rx="1"
      width="64.4661"
      x="0.5"
      y="0.125"
    />
  </mask>
  <rect
    fill="white"
    height="46.6852"
    mask="url(#path-1-inside-1)"
    rx="1"
    stroke="#101B24"
    strokeWidth="4"
    width="64.4661"
    x="0.5"
    y="0.125"
  />
  <path
    d="M6.87939 23.0833H58.5866"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M6.87939 28.9031H58.5866"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M6.87939 34.499H58.5866"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M6.87939 39.7593H58.5866"
    stroke="#101B24"
    strokeWidth="2"
  />
  <rect
    fill="#75DBE7"
    height="5.37217"
    width="19.2503"
    x="6.82349"
    y="6.35156"
  />
  <rect
    fill="#C4F0F5"
    height="3.58145"
    width="38.3886"
    x="6.87939"
    y="14.2417"
  />
  <circle
    cx="63.2873"
    cy="45.2082"
    fill="#75DBE7"
    r="8.2128"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M63.4221 40.3308V49.8146"
    stroke="#101B24"
    strokeWidth="2"
  />
  <path
    d="M68.1643 45.0725L58.6805 45.0725"
    stroke="#101B24"
    strokeWidth="2"
  />
</svg>
`;
